ros_ws/src/arudino/motor/motor.ino

```diff
@@ -106,31 +200,63 @@ void velocityCallback(const std_msgs::Float64& msg)
     }
 }
 
-void setup() 
+/**
+ * @brief The Function will send the updated encoder value to ROS
+ * 
+ * This Function is used to update the main ROS program with
+ * the current number of ticks per timestamp
+ */
+void updateEncoder()
 {
-    //setup pins
-    pinMode(FORWARD_PWM_PIN, OUTPUT);
-    pinMode(REVERSE_PWM_PIN, OUTPUT);
-    pinMode(ENABLE_PIN, OUTPUT);
-
-    //Set Robot to break when starting
-    digitalWrite(ENABLE_PIN, HIGH);
-    analogWrite(FORWARD_PWM_PIN, 0);
-    analogWrite(REVERSE_PWM_PIN, 0);
-
-    //Setup ROS node and topics
-    nodeHandle.initNode();
-    nodeHandle.subscribe(velocitySub);
+    //update the value of the message
+    encoderMessage.ticks = encoder_ticks;
+    encoderMessage.header.stamp = node_handle.now();
+    
+    //publish message
+    encoderPub.publish(&encoderMessage);
 }
 
-void loop() 
+/**
+ * @brief The Function will update the encoder
+ * 
+ * This Function is called whenever there is a change 
+ * to the value of pin 2, it is part of the attachInterrupt routine
+ * It updates the value of encoder_ticks
+ */
+void encoderCount()
 {
-    nodeHandle.spinOnce();
+    encoder_pos_current = digitalRead(ENCODER_PIN); 
+    
+    if ((encoder_pos_last == LOW) && 
+        (encoder_pos_current == HIGH)) 
+    { 
+        encoder_ticks++; 
+    }
+    
+    encoder_pos_last = encoder_pos_current;
```

Comment:
Also this should take into account the direction the wheel is spinning and decrement if it is supposed to be spinning backwards.